calculator_test.go

```diff
@@ -2,23 +2,293 @@ package calculator_test
 
 import (
 	"calculator"
+	"math"
+	"math/rand"
 	"testing"
+	"time"
 )
 
-func TestAdd(t *testing.T) {
+func closeEnough(a, b, epsilon float64) bool {
+	if math.Abs(a-b) < epsilon {
+		return true
+	}
+	return false
+}
+
+const testcount = 100
+const testcaseMaxsize = 10
+
+func TestAddRand(t *testing.T) {
+	t.Parallel()
+	seed := rand.NewSource(time.Now().UnixNano())
+	r := rand.New(seed)
```

Comment:
We can do this all in one go, eliminating `seed`.

Reply:
In fact, in case other tests need a randomness source, we might do this at package level:

```go
var random = rand.New(rand.NewSource(time.Now().UnixNano()))
```

We can then refer to `random.Intn()`, for example, rather than `rand.Intn()`.

Reply:
Agreed.